packages/http/src/types.ts

```diff
@@ -32,7 +32,7 @@ export interface IHttpOperationConfig {
 }
 
 export interface IHttpConfig extends IPrismConfig {
-  mock: boolean | IHttpOperationConfig;
+  mock?: boolean | IHttpOperationConfig;
```

Comment:
Isn't `undefined` making the prism (the framework) unpredictable?

Reply:
I agree, this changes it from a boolean/object to three-state-boolean/object and that's pretty complex.

Reply:
I'm also against this; anything is either something or is not. I struggle to imagine what `undefined` means in this context.

Reply:
Actually undefined has a strong foundations.

The algorithm merges more or less those:
1. default config
2. components config
3. cli config
4. http request config

So if we did set mock: false in 3) and did set nothing in 4) then logically 4)'s config should have mock undefined (meaning we don't want to override state).

But I you were confused by that change that means that sth is wrong, I believe I should introduce Partial<Config> in some places to emphasize that these are partials.